File: src/main/java/aula20190805/estado_das_instâncias_oo/ContaCorrente.java
package aula20190805.estado_das_instâncias_oo;

public class ContaCorrente {
	private float totalDeSaques;
	private float saldoAtual;
	
	public ContaCorrente(float saldoInicial) {
		saldoAtual = saldoInicial;
	}
	
	public void sacar(float valor) {
		this.totalDeSaques += valor;
		this.saldoAtual -= valor;
	}
	
	public float getSaldoAtual() {
		return saldoAtual;
	}
	

}
 

File: src/main/java/aula20190805/estado_das_instâncias_estruturado/conta_corrente.cpp
#include <iostream>

typedef struct ContaCorrente {
	float saldoAtual = 0.00f;
	float totalDeSaques = 0.00f;
} ContaCorrente;

void sacar(ContaCorrente* cc, float valor);

int main(int argc, char** argv) {
	ContaCorrente* cc01 = new ContaCorrente;
	cc01->saldoAtual = 100.00f;

	sacar(cc01, 20.00f);
	
	cc01->saldoAtual = -999.00f;
	
	printf("%f", cc01->saldoAtual);
	
	return 0;
}

void sacar(ContaCorrente* cc, float valor) {
	cc->totalDeSaques += valor;
	cc->saldoAtual -= valor;
}
